pkg/v1/webhooks/certificates.go

```diff
@@ -63,53 +63,70 @@ func addCertsToWebhookConfigs(ctx context.Context, k8sclient kubernetes.Interfac
 	return nil
 }
 
-func NewTLSSecret(ctx context.Context, secretName, serviceName, certPath, keyPath, namespace string) (*corev1.Secret, error) {
-	secret, err := resources.MakeSecretInternal(ctx, secretName, namespace, serviceName)
-	if err != nil {
-		return nil, err
-	}
+func WriteServerTLSToFileSystem(ctx context.Context, certPath, keyPath string, secret *corev1.Secret) error {
+	// Writes servers certificate and key in provided secret to the filesystem paths provided
 
 	if err := certutil.WriteCert(certPath, secret.Data[resources.ServerCert]); err != nil {
-		return secret, err
+		return err
 	}
 	if err := keyutil.WriteKey(keyPath, secret.Data[resources.ServerKey]); err != nil {
-		return secret, err
+		return err
 	}
-
-	return secret, nil
+	return nil
 }
 
 func InstallNewCertificates(ctx context.Context, k8sConfig *rest.Config, certPath, keyPath, secretName, namespace, serviceName, labelSelector string) (*corev1.Secret, error) {
+	// Creates a new set of keys and certificates and saves them to the filesystem paths provided.
+	// Adds the CA certificate to webhook configurations matching label selector.
+	// Returns a secret containing the server key, sever certificate and CA certificate.
+
 	if labelSelector == "" {
 		return nil, fmt.Errorf("label selector not provided for webhook configurations udpate")
 	}
-	secret, err := NewTLSSecret(ctx, secretName, serviceName, certPath, keyPath, namespace)
+	secret, err := resources.MakeSecret(ctx, secretName, namespace, serviceName)
 	if err != nil {
```

Comment:
Why is this lines not covered as part of test ? Seems like this is important functionality that we want to test

Reply:
The calling function requires a cluster to be running.
The lines are actually covers in testing here: https://github.com/vmware-tanzu/tanzu-framework/pull/2417/files#diff-447441730f38027eb4773a493cd8ed684fa7ef1a7766f8f0f8d7184f41f3be3aR34